### PR TITLE
FLUT-6424 - [Others] Added the UG for integrate SfSignature Pad with SfPdfViewer.

### PR DESCRIPTION
**Pull request by deepika134**
## Description ##

 Added the UG for integrate the SignaturePad signatures to Syncfusion Flutter PdfViewer.

### CODE REVIEW

**Comment by deepika134** [comment]
rebuild

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+layout: post
+title: Add SfSignaturePad signature in Flutter PDF Viewer | Syncfusion
+description: Learn here all about how to add SfSignaturePad signatures in the Syncfusion Flutter PDF Viewer (SfPdfViewer) widget and more.
+platform: Flutter
+control: SfPdfViewer
+documentation: ug
+---
+
+# How to add SfSignaturePad signature in SfPdfViewer?
```
Comment (dillibabu1802) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:10]: Title  - Add digital signature in SfPdfViewer using SfSignaturePad

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+In the `_handleSigningProcess()` method, initially convert the signature drawn in the SfSignaturePad as an image using the [toImage()](https://pub.dev/documentation/syncfusion_flutter_signaturepad/latest/signaturepad/SfSignaturePadState/toImage.html) method in SfSignaturePad .Then create a [PdfSignatureField()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfSignatureField-class.html) and drew the signature image in the PDF document using the [drawImage()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfGraphics/drawImage.html) method in `Syncfusion PDF Library` and then saved the document as bytes and then loaded those bytes using the [SfPdfViewer.memory()](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/SfPdfViewer/SfPdfViewer.memory.html). The following code example explains the same.
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:14]: In the `_handleSigningProcess()` method, initially convert the signature drawn in the SfSignaturePad as an image using the [toImage()](https://pub.dev/documentation/syncfusion_flutter_signaturepad/latest/signaturepad/SfSignaturePadState/toImage.html) method in SfSignaturePad .Then create a [PdfSignatureField()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfSignatureField-class.html) and drew the signature image in the PDF document using the [drawImage()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfGraphics/drawImage.html) method in `Syncfusion PDF Library` and then saved the document as bytes and then loaded those bytes using the [SfPdfViewer.memory()](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/SfPdfViewer/SfPdfViewer.memory.html). The following code example explains the same.

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    final bytes = await data.toByteData(format: ui.ImageByteFormat.png);
+
+    final ByteData docBytes = await rootBundle.load("assets/sample.pdf");
+    final Uint8List documentBytes = docBytes.buffer.asUint8List();
+    ByteData certBytes = await rootBundle.load("assets/certificate.pfx");
+    final Uint8List certificateBytes = certBytes.buffer.asUint8List();
+
+    PdfDocument document = PdfDocument(inputBytes: documentBytes);
+
+    //Gets the first page of the document
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:71]: //Get the first page of the document

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    PdfPage page = document.pages[0];
+
+    //Creates a digital signature and sets signature information.
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:74]: //Create a digital signature and set the signature information.

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    PdfSignatureField _signatureField = PdfSignatureField(page, 'signature',
+        bounds: const Rect.fromLTRB(300, 500, 550, 700),
+        signature: PdfSignature(
+            //Creates a certificate instance from the PFX file with a private key.
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:78]: //Create a certificate instance from the PFX file with a private key.

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+            certificate: PdfCertificate(certificateBytes, 'password123'),
+            contactInfo: 'johndoe@owned.us',
+            locationInfo: 'Honolulu, Hawaii',
+            reason: 'I am author of this document.',
+            digestAlgorithm: DigestAlgorithm.sha256,
+            cryptographicStandard: CryptographicStandard.cms));
+
+    //Gets the signature field appearance graphics.
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:86]: //Get the signature field appearance graphics.

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    PdfGraphics? graphics = _signatureField.appearance.normal.graphics;
+
+    //Draws the signature image in PDF page.
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:89]: //Draw the signature image in the PDF page.

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    graphics?.drawImage(PdfBitmap(bytes!.buffer.asUint8List()),
+        const Rect.fromLTWH(0, 0, 250, 200));
+
+    //Add a signature field to the form.
+    document.form.fields.add(_signatureField);
+
+    //Flattens the PDF form field annotation
```
Comment (carolinedolly) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:96]: //Flatten the PDF form field annotation

**Review by carolinedolly — APPROVED** [review]

**Review by dillibabu1802 — CHANGES_REQUESTED** [review]

File: Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+In the `_handleSigningProcess()` method, save the signature as image in SfSignaturePad using the [toImage()](https://pub.dev/documentation/syncfusion_flutter_signaturepad/latest/signaturepad/SfSignaturePadState/toImage.html) method.Then create a [PdfSignatureField()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfSignatureField-class.html) and add the signature image as digital signature in the PDF document using the [drawImage()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfGraphics/drawImage.html) method in `Syncfusion PDF Library` and then saved the document as bytes and then loaded those bytes using the [SfPdfViewer.memory()](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/SfPdfViewer/SfPdfViewer.memory.html). The following code example explains the same.
```
Comment (dillibabu1802) [inline Flutter/pdf-viewer/How to/Add SignaturePad signatures to Syncfusion Flutter PdfViewer.md:14]: In the `_handleSigningProcess()` method, the signature in SfSignaturePad is saved as image using the [toImage()](https://pub.dev/documentation/syncfusion_flutter_signaturepad/latest/signaturepad/SfSignaturePadState/toImage.html) method. Create a [PdfSignatureField()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfSignatureField-class.html) and add the signature image as digital signature in the PDF document using the [drawImage()](https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfGraphics/drawImage.html) method in `Syncfusion PDF Library` and then save the document as bytes. Load the saved document bytes using the [SfPdfViewer.memory()](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/SfPdfViewer/SfPdfViewer.memory.html). The following code example explains the same.

**Review by dillibabu1802 — APPROVED** [review]
Approved

**Pull request merged**